src/plugins/vis_types/vega/public/vega_view/vega_base_view.js

```diff
@@ -180,16 +211,9 @@ export class VegaBaseView {
         // because user can only supply pure JSON data structure.
         uri = uri.url;
       } else if (!this._enableExternalUrls) {
-        throw new Error(
-          i18n.translate('visTypeVega.vegaParser.baseView.externalUrlsAreNotEnabledErrorMessage', {
-            defaultMessage:
-              'External URLs are not enabled. Add   {enableExternalUrls}   to {kibanaConfigFileName}',
-            values: {
-              enableExternalUrls: 'vis_type_vega.enableExternalUrls: true',
-              kibanaConfigFileName: 'kibana.yml',
-            },
-          })
-        );
+        this.handleExternalUrlError(getExternalUrlsAreNotEnabledError());
+      } else if (this._externalUrl.validateUrl(uri)) {
```

Comment:
the current version is actually preventing the allowed URL to pass.
```suggestion
      } else if (!this._externalUrl.validateUrl(uri)) {
```

this line was preventing the default behaviour of no defined `policy` to open a URL, and inverting the `allow` behaviour when defined.